components/collector/src/source_collectors/gitlab/base.py

```diff
@@ -62,6 +63,20 @@ async def _api_url(self) -> URL:
         """Override to return the jobs API."""
         return await self._gitlab_api_url("jobs")
 
+    async def _next_urls(self, responses: SourceResponses) -> list[URL]:
+        """Return the next (pagination) links from the responses as long as we're within lookback days."""
+        # Note: the GitLab documentation (https://docs.gitlab.com/ee/api/jobs.html#list-project-jobs) says:
+        # "Jobs are sorted in descending order of their IDs." The API has no query parameters to sort jobs by date
+        # created or by date run, so we're going to assume that descending order of IDs is roughly equal to descending
+        # order of date created and date run. As soon as all jobs on a page have a build date that is outside the
+        # lookback period we stop the pagination.
+        lookback_date = date.today() - timedelta(days=int(cast(str, self._parameter("lookback_days"))))
+        for response in responses:
+            for job in await response.json():
```

Comment:
This syntax would suggest using `async for` instead? Perhaps even yielding so this can be used as context manager?

Reply:
No, because `await response.json()` does not return an async iterator.